tools/cdk-build-tools/lib/os.ts

```diff
@@ -3,20 +3,26 @@ import fs = require('fs');
 import util = require('util');
 import { Timers } from "./timer";
 
+export interface ShellOptions {
+  timers?: Timers;
+  env?: {[key: string]: string};
+}
+
 /**
  * A shell command that does what you want
  *
  * Is platform-aware, handles errors nicely.
  */
-export async function shell(command: string[], timers?: Timers): Promise<string> {
-  const timer = (timers || new Timers()).start(command[0]);
+export async function shell(command: string[], options: ShellOptions = {}): Promise<string> {
+  const timer = (options.timers || new Timers()).start(command[0]);
 
   await makeShellScriptExecutable(command[0]);
 
   const child = child_process.spawn(command[0], command.slice(1), {
     // Need this for Windows where we want .cmd and .bat to be found as well.
     shell: true,
-    stdio: [ 'ignore', 'pipe', 'inherit' ]
+    stdio: [ 'ignore', 'pipe', 'inherit' ],
+    env: { ...process.env, ...(options.env || {}) }
```

Comment:
Doc should mention that `options.env` is additive on top of `process.env`.
I don't believe this is the usual behavior (typically I'd expect it uses `process.env` if I don't pass anything, and whatever I passed otherwise).